plugins/guests/redhat/cap/osvariant.rb

```diff
@@ -3,13 +3,12 @@ module GuestRedHat
     module Cap
       class OsVariant
         def self.os_variant(machine)
-          machine.communicate.sudo("grep VARIANT_ID /etc/os-release") do |type, data|
-            if type == :stderr
-              @env.ui.error(data)
-              exit 126
+          if machine.communicate.test("grep VARIANT_ID /etc/os-release")
+            machine.communicate.execute("grep VARIANT_ID /etc/os-release") do |type, data|
```

Comment:
@navidshaikh we can do 
if machine.communicate.test("grep CDK /etc/os-release")  return cdk
elseif 
machine.communicate.test("grep ADB /etc/os-release")  return ADB